src/rules/prefer-istanbul-ignore-reason.ts

```diff
@@ -0,0 +1,53 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { createRule } from './utils';
+
+function hasIgnore(comment: string): boolean {
+  return /istanbul ignore \w+/u.test(comment);
+}
+
+function hasReason(comment: string): boolean {
+  return /istanbul ignore \w+: .*/u.test(comment);
```

Comment:
```suggestion
  return /^istanbul ignore \w+: .*/u.test(comment);
```

Reply:
Istanbul ignores are not required to have a colon; you can use any non-word char including whitespace: `/* istanbul ignore next as it's only required to please typescript */` & `// istanbul ignore if - we've not yet got a nice way to test this branch`.

Reply:
I found an example with a colon in Istanbul docs and considered it to be a good practice from their side. You're right. This rule should not enforce this.